.github/workflows/python.yml

```
@@ -129,74 +157,59 @@ jobs:
               continue-on-error: true
               uses: actions/upload-artifact@v4
               with:
-                  name: test-report-python-${{ matrix.python }}-${{ matrix.engine.type }}-${{ matrix.engine.version }}-${{ matrix.host.RUNNER }}
+                  name: test-report-python-${{ matrix.python }}-${{ matrix.engine.type }}-${{ matrix.engine.version }}-${{ matrix.host.RUNNER }}-${{ matrix.python }}
                   path: |
                       python/python/tests/pytest_report.html
                       utils/clusters/**
                       benchmarks/results/**
-
-    test-pubsub:
+    test-pubsub-python:
         runs-on: ${{ matrix.host.RUNNER }}
-        needs: load-engine-matrix
+        needs: [create-version-matrix, get-matrices]
         timeout-minutes: 35
         strategy:
             fail-fast: false
             matrix:
-                engine: ${{ fromJson(needs.load-engine-matrix.outputs.matrix) }}
-                python:
-                  # - "3.8"
-                  # - "3.9"
-                  # - "3.10"
-                  # - "3.11"
-                    - "3.12"
-                host:
-                    - {
-                        OS: ubuntu,
-                        RUNNER: ubuntu-latest,
-                        TARGET: x86_64-unknown-linux-gnu
-                    }
-                    # - {
-                    #     OS: macos,
-                    #     RUNNER: macos-latest,
-                    #     TARGET: aarch64-apple-darwin
-                    # }
-
+                engine: ${{ fromJson(needs.get-matrices.outputs.engine-matrix-output) }}
+                python: ${{ fromJson(needs.create-version-matrix.outputs.python-matrix-output) }}
+                host: ${{ fromJson(needs.get-matrices.outputs.host-matrix-output)}}
```

You have an error
```
Error when evaluating 'strategy' for job 'test-pubsub-python'. .github/workflows/python.yml (Line: 174, Col: 23): A mapping was not expected
```
probably this fixes:
```suggestion
                host: ${{ fromJson(needs.get-matrices.outputs.host-matrix-output).include }}
```

---

Please track CI jobs for errors